core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1699,6 +1699,21 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     withAttributeNamespaceCheck(attrNames)(op)
   }
 
+  private def createWorkflowCollectionForWorkspace(workspaceId: String) = {
+    for {
+      workspacePolicies <- samDAO.listPoliciesForResource(SamResourceTypeNames.workspace, workspaceId, userInfo)
+      policyMap = workspacePolicies.map(pol => pol.policyName -> pol.email).toMap
+      _ <- samDAO.createResource(SamResourceTypeNames.workflowCollection, workspaceId, userInfo)
+      _ <- samDAO.overwritePolicy(SamResourceTypeNames.workflowCollection, workspaceId, SamWorkflowCollectionPolicyNames.workflowCollectionOwnerPolicyName,
+        SamPolicy(Set(policyMap(SamWorkspacePolicyNames.projectOwner), policyMap(SamWorkspacePolicyNames.owner)), Set.empty, Set(SamWorkflowCollectionRoles.owner)), userInfo)
+      _ <- samDAO.overwritePolicy(SamResourceTypeNames.workflowCollection, workspaceId, SamWorkflowCollectionPolicyNames.workflowCollectionWriterPolicyName,
+        SamPolicy(Set(policyMap(SamWorkspacePolicyNames.writer)), Set.empty, Set(SamWorkflowCollectionRoles.writer)), userInfo)
+      _ <- samDAO.overwritePolicy(SamResourceTypeNames.workflowCollection, workspaceId, SamWorkflowCollectionPolicyNames.workflowCollectionReaderPolicyName,
+        SamPolicy(Set(policyMap(SamWorkspacePolicyNames.reader)), Set.empty, Set(SamWorkflowCollectionRoles.reader)), userInfo)
```

Comment:
can these 3 overwrite requests be done in parallel?

Reply:
These policies should have members. The writer policy should have the email address of the workspace writer policy, the reader policy should have the email address of the workspace reader policy. The owner policy should have the email address of the workspace owner and project owner policies. I don't think we need a separate project owner policy on workflow collections.

Reply:
createResourceFull was a pain for some reason but I can take another look.

re adding the workspace policy members to the collection policies -- isn't that what passing in `SamPolicy(Set(policyMap(SamWorkspacePolicyNames.reader` does? The values of `policyMap` are the email addresses of the workspace policies (see line 1705).

Reply:
yup, sorry, misread that... there are too many sets

but createResourceFull is the right thing. Not only is it fewer sam api calls, but createResource will also create a default policy for you which you don't want.